### DISCUSSION TITLE
MQTT Connector

### DISCUSSION THREAD

**Discussion by tobehn** [Budibase/budibase#2567] [Ideas]
As once mentioned in a poll like discussion here https://github.com/Budibase/budibase/discussions/891#discussioncomment-344372 i would like to bring this up again. 

It would be huge to connect to a MQTT Broker and gain realtime data from any connected device/application.
If this data could then selectively be stored into DB or used for further automation or calculation it could become very powerful.

Perhaps this could be helpful:
https://github.com/4refr0nt/svelte-mqtt-monitor

**Comment by mjashanks** [comment]
They @tobehn. I really like this idea. To be honest, I don't think we are going to be doing this soon.

If we were to implement this, it would make the most sense to do so via an Automation trigger... i.e. `MQTT Trigger`. In doing so, we would have to start opening this up to every other MQ as well and make triggers be pluggable. 

I think this is a really nice idea that could open a lot of uses, but I can't see this being a priority at this moment in time. There is always the opportunity for things like this to be prioritised if we have a bigger, paying customer requesting it (happy to talk if you might fit into that category!). 

One day, I hope that we will have this. The workaround, for now, is to 

1. Create an automation, with a Webhook trigger
2. Run an MQTT handler to feed requests to this webhook

**Comment by MichaelUray** [comment]
I am also interested to have MQTT implemented, maybe this JS library could help:
https://github.com/mqttjs/MQTT.js

**Reply by poirazis** [reply]
I have created an MQTT automation step that can publish to an MQTT broker but custom automations are only available for self hosted instances. Its essentially a wrapper to the MQTT.js
